### ISSUE TITLE
Howto - Changing taxes should not affect order history

### ISSUE THREAD

**Issue by gabiudrescu** [Sylius/Sylius#7167]
In the good ol' tradition, in Romania every 31st of December the gov. decides to change the VAT. last year from 24% to 20% except food and non-alcoholic beverages that got 9%. this year is going to become 19% - the general one. 

While last year we weren't using Sylius in production, now this change affects our team too. I tried it on our stage environment and I have realized that orders items are tied to taxes by reference. Which means if I change a certain tax in a tax category, all orders (tax) amount change.

Which is a nightmare, from reporting and accounting point of view. Imagine you take a look at last year's numbers and they suddenly became smaller because of a tax reduction that should only apply to this year - not retroactively. 

We're lucky that we take a snapshot of orders in a JSON and send it downstream, without any reference, but I'm wondering how others handle this.

Imho, data must be copied to the order:
 - product name
 - product price
 - product relevant attributes (e.g. options)
 - product taxes
 - adjustments
 - addresses
 - shipment & payment options

Right now, anything that is referencing data it makes me feel uncertain about how the system is going to behave when we have some changes.

**Comment by pamil** [comment]
This is something we should definitely deal with before beta 👍

**Comment by Thalvik** [comment]
This is usually solved by referencing taxes by date, not by slug or id. So in taxes table you will have for each tax the date from and date to this tax is valid. And when querying for lets say orders or some other entity connected with taxes, you take the order creation date and return tax based on that date. This can also be temp fix until they add support for this in release. @gabiudrescu

**Comment by gabiudrescu** [comment]
as I said, we are currently not affected by this, as we are using Sylius
only for customer front end. after order placement, we take a snapshot of
the order + related entities, tax included, and we ship it somewhere else
as a JSON.

my question is not about how this could be treated, but rather about how it
is currently treated by other Sylius users.

**Comment by michalmarcinkowski** [comment]
@gabiudrescu but after the order is placed it should not be modified at all, how is it possible that all orders were recalculated after tax rate change?

**Comment by gabiudrescu** [comment]
@michalmarcinkowski I think I rushed a little bit when I posted this. it seems only affects our instance of Sylius - we are running 0.19.

In demo.sylius.org the behavior described by me no longer reproduces - changing the tax amount no longer changes the taxes on orders that have already been placed.

----

But, I stumbled upon other some sort of problems on current demo.sylius.org:
 - a shipment method cannot be deleted if it is used in an order, but a payment order can be. then, an order remains without any method for the payment:
![image](https://cloud.githubusercontent.com/assets/5156054/21578308/d5504e32-cf83-11e6-8647-d6b8b94789da.png)
vs
![image](https://cloud.githubusercontent.com/assets/5156054/21578310/edd3d794-cf83-11e6-8f30-9d3ff8361426.png)


 - changing the amount of a tax it will not change the total of a cart unless you go through address / payment step again. refreshing in the last checkout step is not changing the amount
